runtime/src/accounts_db.rs

```diff
@@ -672,25 +681,56 @@ impl AccountsDB {
     // can be purged because there are no live append vecs in the ancestors
     pub fn clean_accounts(&self) {
         self.report_store_stats();
-        let mut purges = HashMap::new();
-        let mut purges_in_root = Vec::new();
+
         let no_ancestors = HashMap::new();
+        let mut accounts_scan = Measure::start("accounts_scan");
         let accounts_index = self.accounts_index.read().unwrap();
+        let pubkeys: Vec<Pubkey> = accounts_index.account_maps.keys().cloned().collect();
+        // parallel scan the index.
+        let (mut purges, purges_in_root) = pubkeys
+            .par_chunks(4096)
+            .map(|pubkeys: &[Pubkey]| {
+                let mut purges_in_root = Vec::new();
+                let mut purges = HashMap::new();
+                for pubkey in pubkeys {
+                    if let Some((list, index)) = accounts_index.get(pubkey, &no_ancestors) {
+                        let (slot, account_info) = &list[index];
+                        if account_info.lamports == 0 {
+                            purges.insert(*pubkey, accounts_index.would_purge(pubkey));
+                        } else if accounts_index.uncleaned_roots.contains(slot) {
+                            purges_in_root.push(*pubkey);
+                        }
+                    }
+                }
+                (purges, purges_in_root)
+            })
+            .reduce(
+                || (HashMap::new(), Vec::new()),
+                |m1, m2| {
+                    // Collapse down the hashmaps/vecs into one.
+                    let x = m2.0.iter().fold(m1.0, |mut acc, (k, vs)| {
+                        let entry = acc.entry(k.clone()).or_insert_with(|| vec![]);
+                        entry.extend(vs.clone());
```

Comment:
I think `acc.insert(k.clone(), vs.clone())` is just suffice because no matter how `pubkey` are grouped and parallelized, there should always be only one `HashMap` containing a given `pubkey` even under the associative property requirement for the `.reduce()`.